projects/vdk-core/tests/functional/ingestion/test_ingest_chain.py

```diff
@@ -9,95 +9,227 @@
 
 
+# INGEST_PAYLOAD_PREPROCESS_SEQUENCE=A,B and VDK_INGEST_METHOD_DEFAULT=C and
+#    job code is D
 @mock.patch.dict(
     os.environ,
-    {"INGEST_PAYLOAD_PREPROCESS_SEQUENCE": "convert-to-string,add-payload-size"},
+    {
+        "VDK_INGEST_PAYLOAD_PREPROCESS_SEQUENCE": "convert-to-string,"
+        "add-payload-size",
+        "VDK_INGEST_METHOD_DEFAULT": "dummy-ingest",
+    },
 )
-def test_chained_ingest_direct_method_passed():
-    pre_ingest_plugin = ConvertPayloadValuesToString()
+def test_chained_ingest_no_direct_method_passed():
+    payload_to_string_plugin = ConvertPayloadValuesToString()
     payload_size_plugin = AddPayloadSizeAsColumn()
+    dummy_ingest_plugin = DummyIngestionPlugin()
     ingest_plugin = IngestIntoMemoryPlugin()
     runner = CliEntryBasedTestRunner(
-        pre_ingest_plugin, payload_size_plugin, ingest_plugin
+        payload_to_string_plugin,
+        payload_size_plugin,
+        dummy_ingest_plugin,
+        ingest_plugin,
     )
 
-    # Use a sample data job, in which `method` argument is passed to ingestion
-    # method calls.
-    result: Result = runner.invoke(["run", utils.job_path("test-ingest-job")])
+    # Use a sample data job, in which `method` argument is not passed to
+    # ingestion method calls.
+    result: Result = runner.invoke(
+        ["run", jobs_path_from_caller_directory("test-ingest-job")]
+    )
 
     cli_assert_equal(0, result)
 
-    # TODO: Uncomment below lines when implementation is complete
-    # expected_object = dict(
-    #     int_key=str(42),
-    #     str_key="example_str",
-    #     bool_key=str(True),
-    #     float_key=str(1.23),
-    #     nested=str(dict(key="value")),
-    # )
-    # assert ingest_plugin.payloads[0].payload[0] == expected_object
+    expected_object = dict(
+        int_key=str(42),
+        str_key="example_str",
+        bool_key=str(True),
+        float_key=str(1.23),
+        nested=str(dict(key="value")),
+        payload_size=163,
+    )
+    assert ingest_plugin.payloads[0].payload[0] == expected_object
     assert ingest_plugin.payloads[0].destination_table == "object_table"
 
-    # TODO: Uncomment below lines when implementation is complete
-    # expected_rows_object = {"first": "two", "second": "2"}
-    # assert ingest_plugin.payloads[1].payload[0] == expected_rows_object
+    expected_rows_object = {"first": "two", "payload_size": 80, "second": "2"}
+    assert ingest_plugin.payloads[1].payload[0] == expected_rows_object
     assert ingest_plugin.payloads[1].destination_table == "tabular_table"
+
+
+# INGEST_PAYLOAD_PREPROCESS_SEQUENCE=A,B - A,B are used for pre-processing,
+#    B is used for ingesting and post-processing
+@mock.patch.dict(
+    os.environ,
+    {
+        "VDK_INGEST_PAYLOAD_PREPROCESS_SEQUENCE": "convert-to-string,"
+        "add-payload-size",
+        "VDK_INGEST_METHOD_DEFAULT": "add-payload-size",
+        "VDK_INGEST_PAYLOAD_POSTPROCESS_SEQUENCE": "add-payload-size",
```

Comment:
would removing the post-processing sequence change the test result?

Reply:
Not at this point in time. I've added a TODO to modify the behaviour of the test plugin used for post-processing to expose the result of post-processing, in order for us to verify operations. For the time being, the tests in test_ingester_base handle such checks.